include/fiction/algorithms/simulation/sidb/does_cds_match_logic_for_input_pattern.hpp

```diff
@@ -2,43 +2,52 @@
 // Created by Jan Drewniok on 12.11.24.
 //
 
-#ifndef FICTION_DOES_CHARGE_DISTRIBUTION_MATCH_LOGIC_FOR_INPUT_PATTERN_HPP
-#define FICTION_DOES_CHARGE_DISTRIBUTION_MATCH_LOGIC_FOR_INPUT_PATTERN_HPP
+#ifndef FICTION_DOES_CDS_MATCH_LOGIC_FOR_INPUT_PATTERN_HPP
+#define FICTION_DOES_CDS_MATCH_LOGIC_FOR_INPUT_PATTERN_HPP
 
-#include <fiction/algorithms/simulation/sidb/detect_bdl_wires.hpp>
-#include <fiction/algorithms/simulation/sidb/is_operational.hpp>
-#include <fiction/technology/charge_distribution_surface.hpp>
-#include <fiction/traits.hpp>
+#include "fiction/algorithms/simulation/sidb/is_operational.hpp"
+#include "fiction/technology/charge_distribution_surface.hpp"
+#include "fiction/traits.hpp"
+#include "fiction/types.hpp"
 
 #include <kitty/traits.hpp>
 
+#include <algorithm>
 #include <cassert>
 #include <cstdint>
 #include <vector>
+#include <utility>
 
 namespace fiction
 {
 
 /**
- * This function checks if a charge distribution encodes the correct logic for a specified input pattern
+ * Checks if a given charge distribution correctly encodes the expected logic for a specified input pattern,
  * based on a provided truth table.
  *
+ * @note Kinks are rejected.
+ *
+ * Example:
+ * In the ground state charge distribution of an AND gate, kinks are rejected for the gate to be considered operational.
+ * Given an input pattern of `1`, this function will:
+ * - Verify that the left input wire encodes `0`.
+ * - Verify that the right input wire encodes `1`.
+ * - Verify that the output wire encodes `0`.
+ *
  * @tparam Lyt SiDB cell-level layout type.
  * @tparam TT Truth table type.
  * @param cds Charge distribution surface, containing charge state information for each SiDB.
  * @param params The parameters used to determine if a layout is `operational` or `non-operational`.
  * @param spec Expected Boolean function of the layout given as a multi-output truth table.
- * @param input_pattern The specific input pattern.
- * @param input_bdl_wire Vector of input BDL wires.
- * @param output_bdl_wire Vector of output BDL wires.
- *
+ * @param input_pattern The specific input pattern of the given charge distribution surface.
+ * @param input_and_output_bdl_wire Pair of input and output BDL wires.
```

Comment:
Not sure if it's a good change to make this into an `std::pair` because now it is unclear in which order they go. We have its name as a hint but there is no guarantee.

Reply:
This is a good point, but there is no order guarantee if you do not use a ``std::pair``,  right?

Reply:
That is of course right. If you wanted a strongly-typed solution, you would have to implement some wrapper types like `input_bdl_pair` and `output_bdl_pair`. Which is maybe not the worst idea :thinking: 

The distinct parameter solution is still better than `std::pair` for integration with IDEs and C++20's named parameter support.

Reply:
I really like this idea and have already spent some time on it, but at the moment it seems that several functions need some restructuring, especially ``detect_bdl_wires``. Given that, I would suggest not using those wrappers for now, but maybe in the future.